tubular/nominal.py

```diff
@@ -647,15 +694,23 @@ def transform(self, X):
         X : pd.DataFrame
             Transformed input X with levels mapped accoriding to mappings dict.
 
-        """
-
+        """  
+        
+        if self.level:
+            for response_level in self.response_levels:
+                for column in self.columns:
+                    X[column + '_' + response_level] = X[column]
+            drop_columns = self.columns
+            self.columns = self.mapped_columns
```

Comment:
Changing self.columns here would mean that following calls to transform are broken.  We need to at least set it back to original values after transforming data.  Preferable would be not changing transformer attributes during transform call.

Reply:
I agree with this, given that we have;
- self.columns is to be set to columns to encode in init
- mapping generated for column copies that don't exist in the original dataframe
- mapping accessed by iterating over self.columns in `BaseMappingTransform Mixin.transform()`

We can't set self.columns differently in init because we need to learn the response levels from the data if `level = 'all' has been passed`

A far as I can see that leaves us with either _temporarily_ overwriting in transform then setting back, or writing a new method on `BaseMappingTransfomMixin`

Reply:
Done here: https://github.com/lvgig/tubular/pull/67/commits/4b7baab553f7e7489b1d7d7e9e7c6844865ad72c